observability/micrometer/src/main/java/org/wildfly/extension/micrometer/prometheus/PrometheusRegistryDefinitionRegistrar.java

```diff
@@ -0,0 +1,107 @@
+import org.wildfly.subsystem.service.ServiceInstaller;
+
+public class PrometheusRegistryDefinitionRegistrar implements ChildResourceDefinitionRegistrar, ResourceServiceConfigurator {
+    public static final String NAME = "prometheus";
+    public static final PathElement PATH = PathElement.pathElement("registry", NAME);
+
+    public static final Stability FEATURE_STABILITY = Stability.COMMUNITY;
+    public static final ResourceRegistration RESOURCE_REGISTRATION = ResourceRegistration.of(PATH, FEATURE_STABILITY);
+
+    public static final String HTTP_EXTENSIBILITY_CAPABILITY = "org.wildfly.management.http.extensible";
+    public static final SimpleAttributeDefinition CONTEXT = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.CONTEXT, ModelType.STRING)
+            .setRequired(true)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+    public static final AttributeDefinition SECURITY_ENABLED = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.SECURITY_ENABLED, ModelType.BOOLEAN)
+            .setDefaultValue(ModelNode.TRUE)
+            .setRequired(false)
+            .setRestartAllServices()
+            .setAllowExpression(true)
+            .build();
+    public static final Collection<AttributeDefinition> ATTRIBUTES = List.of(CONTEXT, SECURITY_ENABLED);
+
+    private final WildFlyCompositeRegistry wildFlyRegistry;
+
+    public PrometheusRegistryDefinitionRegistrar(WildFlyCompositeRegistry wildFlyRegistry) {
+        this.wildFlyRegistry = wildFlyRegistry;
+    }
+
+    @Override
+    public ManagementResourceRegistration register(ManagementResourceRegistration parent, ManagementResourceRegistrationContext context) {
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(MicrometerSubsystemRegistrar.RESOLVER.createChildResolver(PATH))
+            .addAttributes(ATTRIBUTES)
+            .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+            .withOperationTransformation("add", new UnaryOperator<OperationStepHandler>() {
+                @Override
+                public OperationStepHandler apply(OperationStepHandler handler) {
+                    return (context, operation) -> {
+                        if (context.getProcessType().isHostController()) {
+                            throw MicrometerExtensionLogger.MICROMETER_LOGGER.prometheusNotSupportedOnHostControllers();
+                        }
+                        handler.execute(context, operation);
+                    };
+                }
+            })
+                .build();
+        ManagementResourceRegistration mrr = parent.registerSubModel(
+                ResourceDefinition.builder(RESOURCE_REGISTRATION,
+                        descriptor.getResourceDescriptionResolver()).build());
+        ManagementResourceRegistrar.of(descriptor).register(mrr);
```

Comment:
The io.prometheus module needs to be recorded as an additionalRuntimePackage, as it should only be an optional dep of any existing modules and Galleon needs to know to provision it if this resource is configured.

See https://wildfly.zulipchat.com/#narrow/channel/174184-wildfly-developers/topic/ResourceDefinition.2EregisterAdditionalRuntimePackages.20.2E.2E.2E/near/485906664 for how to do this.